tests/PySys/plugin_apama/environment_apama_plugin.py

```diff
@@ -0,0 +1,130 @@
+
+        # Register routines to cleanup containers and images added during test
+        self.addCleanupFunction(self.cleanup_project)
+
+    def assert_monitor_installed(self, monitor_name, negate=False, abortOnError=False):
+        """Asserts that a monitor with the given name is loaded in apama correlator"""
+        process = self.startProcess(
+            command=self.sudo,
+            arguments=[self.apama_env_cmd, "engine_inspect", "-m", "-r"],
+            stdouterr="apama_list_monitors"
+        )
+
+        self.assertGrep(
+            process.stdout, monitor_name, contains=not negate, abortOnError=abortOnError)
+
+    def assert_project_installed(self, negate=False, abortOnError=False):
+        """Asserts that an apama project is installed or not"""
+        self.assertPathExists(self.apama_project_dir,
+                              exists=not negate, abortOnError=abortOnError)
+
+    def install_project(self, archive_path):
+        """Install apama project from the provided archive."""
+        self.startProcess(
+            command=self.sudo,
+            arguments=["mkdir", "-p", self.apama_project_dir],
+            stdouterr="create_project_dir"
+        )
+
+        self.startProcess(
+            command=self.sudo,
+            arguments=["unzip", archive_path, "-d", self.tmp_apama_dir],
+            stdouterr="unzip_apama_project"
+        )
+
+        self.startProcess(
+            command=self.sudo,
+            arguments=["mv", self.tmp_apama_dir, self.apama_project_dir],
+            stdouterr="move_apama_project"
+        )
+
+        self.startProcess(
+            command=self.sudo,
+            arguments=["service", "apama", "restart"],
+            stdouterr="restart_apama_service"
+        )
+
+    def remove_project(self):
+        """Remove the installed apama project, if one is found.
+        """
+        if os.path.exists(self.apama_project_dir):
+            self.log.info(
+                f"Removing apama project")
+
+            self.startProcess(
+                command=self.sudo,
+                arguments=["rm", "-rf", self.apama_project_dir],
+                stdouterr="remove_apama_project"
+            )
+
+            self.startProcess(
+                command=self.sudo,
+                arguments=["service", "apama", "stop"],
+                stdouterr="stop_apama_service"
+            )
+
+    def assert_apama_service_running(self, negate=False):
+        exitStatusExpr = "!=0" if negate else "==0"
+        self.startProcess(
+            command=self.sudo,
+            arguments=["service", "apama", "status"],
+            stdouterr="apama_service_status",
+            expectedExitStatus=exitStatusExpr
+        )
+
+    def plugin_install(self, project_archive_path):
+        """Use apama plugin `install` command to install a project from the provided archive.
+        """
+        self.startProcess(
+            command=self.sudo,
```

Comment:
Does the apama plugin need to run as `sudo`?

Reply:
Yes, because of all the file manipulation that it does in `/etc/tedge` directory and the `service` commands that it invokes.